lldb/source/Plugins/SymbolFile/DWARF/DWARFUnit.h

```diff
@@ -170,7 +170,7 @@ class DWARFUnit : public UserID {
   /// both cases correctly and avoids crashes.
   DWARFCompileUnit *GetSkeletonUnit();
 
-  void SetSkeletonUnit(DWARFUnit *skeleton_unit);
+  bool LinkToSkeletonUnit(DWARFUnit &skeleton_unit);
```

Comment:
Is there a reason this was renamed?

Reply:
It just seemed to me like `Set` does not capture very well the new semantics of the function (a typical setter function just sets the corresponding field and does not second-guess the callers intentions). A nice side benefit is that we can be sure to catch all callers of the api to make sure they're not silently ignoring the result.

 I can revert it if you feel strongly about it.

Reply:
I am fine with the change, makes sense.